### PR TITLE
Update recipe for r-rtassel with symlink

### PR DESCRIPTION
**Pull request by npavlovikj**
Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it. Note that if you are not already a member of the bioconda project (meaning that you can't add this label), please ping `@bioconda/core` so that your PR can be reviewed and merged (please note if you'd like to be added to the bioconda project). Please see https://github.com/bioconda/bioconda-recipes/issues/15332 for more details.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by dpryan79** [comment]
Why is the LD_LIBRARY_PATH needed? Conda should be using an rpath for that.

**Comment by npavlovikj** [comment]
@dpryan79 , all the tests passed here and on my VM, but when I installed "r-rtassel" on our cluster, I got:
```
> library("rTASSEL")
Error: package or namespace load failed for ‘rTASSEL’:
 .onLoad failed in loadNamespace() for 'rJava', details:
  call: dyn.load(file, DLLpath = DLLpath, ...)
  error: unable to load shared object '/home/npavlovikj/.conda/envs/r-rtassel/lib/R/library/rJava/libs/rJava.so':
  libjvm.so: cannot open shared object file: No such file or directory
```
`rJava.so` is looking for `libjvm.so` in `${PREFIX}/lib/server` specifically and therefore I added that to `LD_LIBRARY_PATH`. 

I initially tried symlinking the library as in https://github.com/bioconda/bioconda-recipes/blob/448b7d9aa7328a80f744ee25e017dfeb8eef09cd/recipes/python-bioformats/meta.yaml (and my initial commit in this PR), but that gave the same error. The only thing that worked for me was adding `${PREFIX}/lib/server` to the `LD_LIBRARY_PATH`.

If there is better solution/workaround to this, please let me know - I would be more than happy to adjust the recipe accordingly and correctly.

**Comment by dpryan79** [comment]
Can you see if you can reproduce the original issue in a new environment? I just created one with R 3.6.1 and r-rtassel and got a compatible rJava version.

**Comment by npavlovikj** [comment]
@dpryan79 , I tried creating new environment multiple times, but I kept getting the error. However, I noticed that `r-rjava` from the `r` channel is being used. On the other hand, if I use `r-rjava` from `conda-forge`, I am not getting the error. Looks like `r-rjava` from `conda-forge` is pinned to `openjdk 8.*`. Because I don't use any pinning in this recipe, the newest `openjdk 11.*` is used which uses `r-rjava` from the `r` channel which causes the error.
I guess the best solution would be to add `openjdk 8.*` in this recipe then?

**Comment by dpryan79** [comment]
Please remove the R channel from your list of channels, only conda-forge, bioconda and then defaults should be used.

**Comment by npavlovikj** [comment]
Sorry for not being more clear @dpryan79 , I meant `r` from `defaults`:
```
           channel URLs : https://conda.anaconda.org/hcc/linux-64
                          https://conda.anaconda.org/hcc/noarch
                          https://conda.anaconda.org/t/<TOKEN>/hcc/linux-64
                          https://conda.anaconda.org/t/<TOKEN>/hcc/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/conda-forge/label/cf201901/linux-64
                          https://conda.anaconda.org/conda-forge/label/cf201901/noarch
			  https://conda.anaconda.org/bioconda/linux-64
                          https://conda.anaconda.org/bioconda/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          file:///util/opt/conda_repo/linux-64
                          file:///util/opt/conda_repo/noarch
```
We only support our local channel, conda-forge, bioconda and defaults, and none of the dependencies are pulled from our local channel.

**Comment by duboism** [comment]
This is a bit off-topic but shouldn't such modification of the environment be made in package-specific files (something like `activate_r-tassel.sh` and the corresponding deactivate script) that are installed and uninstalled with `post-link.sh` and `pre-unlink.sh` ? This way they won't interfere with other packages.

**Comment by dpryan79** [comment]
@duboism You can actually put any script under `${PREFIX}/etc/conda/activate.d/` and it will be run upon `conda activate`. So naming the script after the package (and possibly version) would be the optimal solution.

**Comment by dpryan79** [comment]
@npavlovikj Can you show me `conda list` in the affected env?

**Comment by npavlovikj** [comment]
```
[npavlovikj@login.crane npavlovikj]$ conda create -n r-rtassel r-rtassel
[npavlovikj@login.crane npavlovikj]$ conda activate r-rtassel
(r-rtassel) [npavlovikj@login.crane npavlovikj]$ conda list
# packages in environment at /home/npavlovikj/.conda/envs/r-rtassel:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
_r-mutex                  1.0.1               anacondar_1    conda-forge
binutils_impl_linux-64    2.28.1               had2808c_3
binutils_linux-64         7.2.0               had2808c_27
bioconductor-biobase      2.44.0            r36h516909a_1    bioconda
bioconductor-biocgenerics 0.30.0                    r36_1    bioconda
bioconductor-biocparallel 1.18.0            r36he1b5a44_1    bioconda
bioconductor-delayedarray 0.10.0            r36h516909a_1    bioconda
bioconductor-genomeinfodb 1.20.0                    r36_1    bioconda
bioconductor-genomeinfodbdata 1.2.1                     r36_1    bioconda
bioconductor-genomicranges 1.36.0            r36h516909a_1    bioconda
bioconductor-iranges      2.18.1            r36h516909a_0    bioconda
bioconductor-s4vectors    0.22.0            r36h516909a_1    bioconda
bioconductor-summarizedexperiment 1.14.0                    r36_1    bioconda
bioconductor-xvector      0.24.0            r36h516909a_1    bioconda
bioconductor-zlibbioc     1.30.0            r36h516909a_1    bioconda
bwidget                   1.9.11                        0    conda-forge
bzip2                     1.0.8                h516909a_0    conda-forge
ca-certificates           2019.6.16            hecc5488_0    conda-forge
cairo                     1.16.0            hfb77d84_1002    conda-forge
curl                      7.65.3               hf8cf82a_0    conda-forge
fontconfig                2.13.1            h86ecdb6_1001    conda-forge
freetype                  2.10.0               he983fc9_0    conda-forge
fribidi                   1.0.5             h516909a_1002    conda-forge
gcc_impl_linux-64         7.2.0                habb00fd_3
gcc_linux-64              7.2.0               h550dcbe_27
gettext                   0.19.8.1          hc5be6a0_1002    conda-forge
gfortran_impl_linux-64    7.2.0                hdf63c60_3
gfortran_linux-64         7.2.0               h550dcbe_27
glib                      2.58.3            h6f030ca_1002    conda-forge
graphite2                 1.3.13            hf484d3e_1000    conda-forge
gsl                       2.5                  h294904e_0    conda-forge
gxx_impl_linux-64         7.2.0                hdf63c60_3
gxx_linux-64              7.2.0               h550dcbe_27
harfbuzz                  2.4.0                h9f30f68_2    conda-forge
icu                       64.2                 he1b5a44_0    conda-forge
jpeg                      9c                h14c3975_1001    conda-forge
krb5                      1.16.3            h05b26f9_1001    conda-forge
libblas                   3.8.0               11_openblas    conda-forge
libcblas                  3.8.0               11_openblas    conda-forge
libcurl                   7.65.3               hda55be3_0    conda-forge
libedit                   3.1.20181209         hc058e9b_0
libffi                    3.2.1             he1b5a44_1006    conda-forge
libgcc-ng                 9.1.0                hdf63c60_0
libgfortran-ng            7.3.0                hdf63c60_0
libiconv                  1.15              h516909a_1005    conda-forge
liblapack                 3.8.0               11_openblas    conda-forge
libopenblas               0.3.6                h6e990d7_6    conda-forge
libpng                    1.6.37               hed695b0_0    conda-forge
libssh2                   1.8.2                h22169c7_2    conda-forge
libstdcxx-ng              9.1.0                hdf63c60_0
libtiff                   4.0.10            h57b8799_1003    conda-forge
libuuid                   2.32.1            h14c3975_1000    conda-forge
libxcb                    1.13              h14c3975_1002    conda-forge
libxml2                   2.9.9                hee79883_2    conda-forge
lz4-c                     1.8.3             he1b5a44_1001    conda-forge
make                      4.2.1             h14c3975_2004    conda-forge
ncurses                   6.1               hf484d3e_1002    conda-forge
**openjdk                   11.0.1            h516909a_1016    conda-forge**
openssl                   1.1.1c               h516909a_0    conda-forge
pango                     1.42.4               he7ab937_0    conda-forge
pcre                      8.43                 he6710b0_0
pixman                    0.38.0            h516909a_1003    conda-forge
pthread-stubs             0.4               h14c3975_1001    conda-forge
r-assertthat              0.2.1             r36h6115d3f_1    conda-forge
r-backports               1.1.4             r36hcdcec82_1    conda-forge
**r-base                    3.6.1                h6e652e1_3    conda-forge**
r-bh                      1.69.0_1          r36h6115d3f_1    conda-forge
r-bitops                  1.0_6           r36hcdcec82_1003    conda-forge
r-cli                     1.1.0             r36h6115d3f_1    conda-forge
r-colorspace              1.4_1             r36hcdcec82_1    conda-forge
r-crayon                  1.3.4           r36h6115d3f_1002    conda-forge
r-digest                  0.6.20            r36h0357c0b_1    conda-forge
r-dplyr                   0.8.3             r36h0357c0b_1    conda-forge
r-ellipsis                0.2.0.1           r36hcdcec82_1    conda-forge
r-evaluate                0.14              r36h6115d3f_1    conda-forge
r-fansi                   0.4.0           r36hcdcec82_1001    conda-forge
r-formatr                 1.7               r36h6115d3f_1    conda-forge
r-futile.logger           1.4.3           r36h6115d3f_1002    conda-forge
r-futile.options          1.0.1           r36h6115d3f_1001    conda-forge
r-ggplot2                 3.2.0             r36h6115d3f_1    conda-forge
r-glue                    1.3.1             r36hcdcec82_1    conda-forge
r-gtable                  0.3.0             r36h6115d3f_2    conda-forge
r-highr                   0.8               r36h6115d3f_1    conda-forge
r-knitr                   1.23              r36h6115d3f_1    conda-forge
r-labeling                0.3             r36h6115d3f_1002    conda-forge
r-lambda.r                1.2.3           r36h6115d3f_1001    conda-forge
r-lattice                 0.20_38         r36hcdcec82_1002    conda-forge
r-lazyeval                0.2.2             r36hcdcec82_1    conda-forge
r-magrittr                1.5             r36h6115d3f_1002    conda-forge
r-markdown                1.0               r36hcdcec82_1    conda-forge
r-mass                    7.3_51.4          r36hcdcec82_1    conda-forge
r-matrix                  1.2_17            r36hcdcec82_1    conda-forge
r-matrixstats             0.54.0          r36hcdcec82_1001    conda-forge
r-mgcv                    1.8_28            r36hcdcec82_1    conda-forge
r-mime                    0.7               r36hcdcec82_1    conda-forge
r-munsell                 0.5.0           r36h6115d3f_1002    conda-forge
r-nlme                    3.1_140           r36h9bbef5b_1    conda-forge
r-pillar                  1.4.2             r36h6115d3f_2    conda-forge
r-pkgconfig               2.0.2           r36h6115d3f_1002    conda-forge
r-plogr                   0.2.0           r36h6115d3f_1002    conda-forge
r-plyr                    1.8.4           r36h0357c0b_1003    conda-forge
r-purrr                   0.3.2             r36hcdcec82_1    conda-forge
r-r6                      2.4.0             r36h6115d3f_2    conda-forge
r-rcolorbrewer            1.1_2           r36h6115d3f_1002    conda-forge
r-rcpp                    1.0.2             r36h0357c0b_0    conda-forge
r-rcurl                   1.95_4.12         r36hcdcec82_1    conda-forge
r-reshape2                1.4.3           r36h0357c0b_1004    conda-forge
**r-rjava                   0.9_11            r36h96ca727_0**
r-rlang                   0.4.0             r36hcdcec82_1    conda-forge
r-rtassel                 0.1.2019.07.25            r36_0    bioconda
r-scales                  1.0.0           r36h0357c0b_1002    conda-forge
r-snow                    0.4_3           r36h6115d3f_1001    conda-forge
r-stringi                 1.4.3             r36h0e574ca_3    conda-forge
r-stringr                 1.4.0             r36h6115d3f_1    conda-forge
r-tibble                  2.1.3             r36hcdcec82_1    conda-forge
r-tidyselect              0.2.5           r36h0357c0b_1001    conda-forge
r-utf8                    1.1.4           r36hcdcec82_1001    conda-forge
r-vctrs                   0.2.0             r36hcdcec82_1    conda-forge
r-viridislite             0.3.0           r36h6115d3f_1002    conda-forge
r-withr                   2.1.2           r36h6115d3f_1001    conda-forge
r-xfun                    0.8               r36h6115d3f_1    conda-forge
r-yaml                    2.2.0           r36hcdcec82_1002    conda-forge
r-zeallot                 0.1.0           r36h6115d3f_1001    conda-forge
readline                  8.0                  hf8c457e_0    conda-forge
sed                       4.7               h1bed415_1000    conda-forge
tk                        8.6.9             hed695b0_1002    conda-forge
tktable                   2.10                 h555a92e_1    conda-forge
xorg-kbproto              1.0.7             h14c3975_1002    conda-forge
xorg-libice               1.0.10               h516909a_0    conda-forge
xorg-libsm                1.2.3             h84519dc_1000    conda-forge
xorg-libx11               1.6.8                h516909a_0    conda-forge
xorg-libxau               1.0.9                h14c3975_0    conda-forge
xorg-libxdmcp             1.1.3                h516909a_0    conda-forge
xorg-libxext              1.3.4                h516909a_0    conda-forge
xorg-libxrender           0.9.10            h516909a_1002    conda-forge
xorg-renderproto          0.11.1            h14c3975_1002    conda-forge
xorg-xextproto            7.3.0             h14c3975_1002    conda-forge
xorg-xproto               7.0.31            h14c3975_1007    conda-forge
xz                        5.2.4             h14c3975_1001    conda-forge
zlib                      1.2.11            h516909a_1005    conda-forge
zstd                      1.4.0                h3b9ef0a_0    conda-forge

(r-rtassel) [npavlovikj@login.crane npavlovikj]$ R
> library("rTASSEL")
Error: package or namespace load failed for ‘rTASSEL’:
 .onLoad failed in loadNamespace() for 'rJava', details:
  call: dyn.load(file, DLLpath = DLLpath, ...)
  error: unable to load shared object '/home/npavlovikj/.conda/envs/r-rtassel/lib/R/library/rJava/libs/rJava.so':
  libjvm.so: cannot open shared object file: No such file or directory
>
```

**Comment by dpryan79** [comment]
It's unclear to me why you're getting r-rJava from the wrong channel. I perhaps something from conda-forge/label/cf201901 is getting in the way. You might also try setting `channel_priority` to `strict` in `conda config`.

File: recipes/r-rtassel/meta.yaml [hunk 1]
```diff
@@ -31,6 +31,7 @@ requirements:
     - bioconductor-iranges
     - bioconductor-genomicranges
     - bioconductor-summarizedexperiment
+    - openjdk 8.*
```
Comment (bgruening) [inline recipes/r-rtassel/meta.yaml]: is it needed as host requirement?

**Pull request closed without merging**